storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/RocksDbDatabase.java

```diff
@@ -304,13 +318,16 @@ boolean batchVerifyHistoricalBlockSignatures(final Collection<SignedBeaconBlock>
     List<Bytes> signingRoots = new ArrayList<>();
     List<List<BLSPublicKey>> proposerPublicKeys = new ArrayList<>();
 
-    // TODO: This domain is dependent on the fork version. Thus when we support forks, we're going
-    // to have to change the way we retrieve the domain here.
-    Bytes32 domain = get_domain(finalizedState, DOMAIN_BEACON_PROPOSER);
+    final Bytes32 genesisValidatorsRoot = finalizedState.getForkInfo().getGenesisValidatorsRoot();
 
     blocks.forEach(
         signedBlock -> {
-          BeaconBlock block = signedBlock.getMessage();
+          final BeaconBlock block = signedBlock.getMessage();
+          final UInt64 epoch = compute_epoch_at_slot(block.getSlot());
+          final Fork fork = specProvider.fork(epoch);
```

Comment:
I wonder if it's worth having a `forkAtSlot` method that does the conversion to epoch for you.

Reply:
as discussed. need epoch in this instance, but potentially use cases down the line. will leave it for now.